backend/inspirehep/records/tasks.py

```diff
@@ -43,3 +56,49 @@ def update_records_relations(uuids):
 
     db.session.commit()
     return uuids
+
+
+@shared_task(ignore_result=False, bind=True, queue="migrator")
+def recalculate_references_for_merged_record(self, merged_record_uri, record_uri):
+    merged_record_schema = merged_record_uri.rsplit("/", 2)[1]
+    if merged_record_schema == "literature":
+        merged_record_schema = "hep"
+    referenced_records_indices_and_paths = get_refs_to_schemas()[merged_record_schema]
+    match_reference_with_config(
+        referenced_records_indices_and_paths, merged_record_uri, record_uri
+    )
+
+
+def match_reference_with_config(config_list, merged_record_uri, new_record_uri):
+    for index, path in config_list:
+        config = (
+            generate_matcher_config_for_nested_reference_field(index, path)
+            if path in ["authors", "publication_info", "first_author", "supervisors"]
+            else generate_matcher_config_for_reference_field(index, path)
+        )
+        matched_records = match({"$ref": merged_record_uri}, config,)
+        for matched_record in matched_records:
+            uuid = get_value(matched_record, "_id")
+            matched_inspire_record = InspireRecord.get_record(uuid)
+            for referenced_record in get_value(
+                matched_inspire_record, path.rstrip(".$ref"), []
```

Comment:
You shouldn't use `rstrip` for this kind of stuff (it removes individual matching *characters* at the end of the string). In Python 3.9 there's `removesuffix`, but in the meantime you can slice the string
```suggestion
                matched_inspire_record, path[:-len(".$ref")], []
```